java/tests/com/google/privacy/differentialprivacy/BoundedSumTest.java

```diff
@@ -678,4 +680,215 @@ private static void testForBias(
     // equal to the raw count.
     assertThat(stats.mean()).isWithin(sampleTolerance).of(rawEntry);
   }
-}
+
+  @Test
+  public void computeConfidenceInterval_negativeSumBounds() {
+    sum =
+            BoundedSum.builder()
+                    .epsilon(EPSILON)
+                    .delta(DELTA)
+                    .noise(noise)
+                    .maxPartitionsContributed(1)
+                    .lower(-5)
+                    .upper(-1)
+                    .build();
+    when(noise.computeConfidenceInterval(
+            anyDouble(), anyInt(), anyDouble(), anyDouble(), anyDouble(), anyDouble()))
+            .thenReturn(ConfidenceInterval.create(-5,3));
+    sum.computeResult();
+
+    // The result interval = (-5, 3), but it should be clamped to (-5, 0), because the bounds are negative.
+    assertThat(sum.computeConfidenceInterval(0.152145599))
+            .isEqualTo(ConfidenceInterval.create(-5, 0));
+  }
+
+  @Test
+  public void computeConfidenceInterval_positiveSumBounds() {
+    sum =
+            BoundedSum.builder()
+                    .epsilon(EPSILON)
+                    .delta(DELTA)
+                    .noise(noise)
+                    .maxPartitionsContributed(1)
+                    .lower(1)
+                    .upper(5)
+                    .build();
+    when(noise.computeConfidenceInterval(
+            anyDouble(), anyInt(), anyDouble(), anyDouble(), anyDouble(), anyDouble()))
+            .thenReturn(ConfidenceInterval.create(-5,3));
+    sum.computeResult();
+
+    // The result interval = (-5, 3), but it should be clamped to (0, 3), because the bounds are positive.
+    assertThat(sum.computeConfidenceInterval(0.152145599))
+            .isEqualTo(ConfidenceInterval.create(0, 3));
+  }
+
+  @Test
+  public void computeConfidenceInterval_differentSumBoundsSigns() {
+    sum =
+            BoundedSum.builder()
+                    .epsilon(EPSILON)
+                    .delta(DELTA)
+                    .noise(noise)
+                    .maxPartitionsContributed(1)
+                    .lower(-1)
+                    .upper(5)
+                    .build();
+    when(noise.computeConfidenceInterval(
+            anyDouble(), anyInt(), anyDouble(), anyDouble(), anyDouble(), anyDouble()))
+            .thenReturn(ConfidenceInterval.create(-5,3));
+    sum.computeResult();
+
+    // The result interval = (-5, 3) and it should not be clamped.
+    assertThat(sum.computeConfidenceInterval(0.152145599))
+            .isEqualTo(ConfidenceInterval.create(-5, 3));
+  }
+
+  @Test
+  public void computeConfidenceInterval_infiniteSumBounds() {
```

Comment:
Is this really behaviour that we want. I am notsure we need/should be testing for this

Reply:
Hmm, in the new version i also added _clampsNegativeInterval, so overall what this does is just testing that NEGATIVE_INFINITY gets clamped to 0.0. I am not sure of whether or not i should keep this test.  @MashaTelyatnikova WDYT?